### PR TITLE
Fix/usoclient

### PR DESCRIPTION
**Pull request by niStee**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] I have read `CONTRIBUTING.md`
- [x] The code compiles (`cargo build`)
- [x] The code passes rustfmt (`cargo fmt`)
- [x] The code passes clippy (`cargo clippy`)
- [x] The code passes tests (`cargo test`)
- [ ] *Optional:* I have tested the code myself
 
## For new steps
- [ ] *Optional:* Topgrade skips this step where needed
- [ ] *Optional:* The `--dry-run` option works with this step
- [ ] *Optional:* The `--yes` option works with this step if it is supported by 
  the underlying command

If you developed a feature or a bug fix for someone else and you do not have the
means to test it, please tag this person here.

### CODE REVIEW

File: src/steps/os/windows.rs [hunk 1]
```diff
@@ -212,6 +214,14 @@ pub fn windows_update(ctx: &ExecutionContext) -> Result<()> {
         return powershell.windows_update(ctx);
     }
 
+    let info: os_info::Info = os_info::get();
+    let edition: String = info.edition().unwrap_or("Unknown").to_string();
+    if edition.contains("Windows 11") {
+        print_separator("Windows Update");
+        println!("Windows 11 does not support Windows Update via usoclient.exe. Consider installing PSWindowsUpdate");
```
Comment (SteveLauC) [inline src/steps/os/windows.rs:221]: Based on [my previous test](https://github.com/topgrade-rs/topgrade/issues/482#issuecomment-1643109218), this `usoclient` command does not work on Windows 10 either

**Comment by SteveLauC** [comment]
Thanks for your contribution! For these 3 PRs, if you want to update the dependencies, please consider doing it in a separate PR

File: src/steps/os/windows.rs (hunk 1 above)
Comment (niStee) [reply src/steps/os/windows.rs:221]: In this case, it is recommended to completely remove usoclient. Previous versions of Windows (7/8), which may still support it, have reached end of life.

Comment (SteveLauC) [reply src/steps/os/windows.rs:221]: Makes sense to me

**Review by SteveLauC — APPROVED** [review]
Thanks!

**Pull request merged**